### ISSUE TITLE
Make entities default unlikeable

### ISSUE THREAD

**Issue by jdalsem** [Elgg/Elgg#5996]
By default all entities are likeable. It would make more sense to have them not likeable and make them likeable with a canLike hook

**Comment by mrclay** [comment]
I'm not against this but it's a breaking change and will have to wait for 2.0. We'd also need to implement hooks for content in the bundled plugins.

**Comment by iionly** [comment]
Kind of a pity that this little BC breaking change will require separate releases for 1.X and 2.X for a lot of plugins while ALL the other changes of 2.0 (with exception of making discussions a separate plugins) still allowed to make a single release for 1.X and 2.X.

**Comment by jdalsem** [comment]
@iionly thats why it is a breaking change, however, you could still make 1 version that works for both (if this is the only reason), because you can have the same hook registration in 1.x.

**Comment by ewinslow** [comment]
But wait of you guard the function call with a conditional to check if it exists, won't that work across 1.x and 2.x?

**Comment by jdalsem** [comment]
> But wait of you guard the function call with a conditional to check if it exists, won't that work across 1.x and 2.x?

you were talking to @iionly right?

**Comment by ewinslow** [comment]
Yea

**Comment by iionly** [comment]
Well, I've got a little bit frustrated this morning about the last minute BC breaking changes that foiled my plan to have compatible versions of all my plugins ready at about the end of next week allowing me to take a little break to enjoy summer. Now I will have to figure out again which plugins I have to rework again that I thought until yesterday would already be 2.0-ready. And I kind of lost the motivation to do that just now. So, I think I will rather take my break just now not putting me under any pressure anymore with getting the plugins working on 2.0 as soon as possible.

@jdalsem it seemed to me this morning that the alternative approach of Steve would make it over the plugin hook approach. And I was not keen on using function_exist just to have a plugin version that works both on 1.X and 2.X. And even with the plugin hook approach having been selected a new plugin release will be necessary nevertheless. Almost all other BC breaking issues of 2.0 wouldn't require a new plugin release if the plugin is already fully 1.10/1.11 compatible without any deprecation issues and all JS code AMDed (which was basically the only thing I had to work on so far).
